### PR TITLE
Sprint 3/issue 34#add flatlist component

### LINKED ISSUE #34: Add FlatList component
## REQUIREMENTS / DESCRIPTION

This component will be used for the podcasts, music and reading screens. It should be able to ListeningCards components inside and offer responsive scrolling for these screens

### PR DESCRIPTION
**Pull request by Lizzflores**
# Description

What did you implemented/Why did you implemented this:

This pull request includes a new 'CardList' component, designed to provide a unified way of displaying different types of cards, including podcasts, music, and stories. The component uses the FlatList component to ensure efficient rendering and optimal scrolling performance.

### CODE REVIEW

File: languageApp/src/components/CardList/CardList.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  const route = useRoute();
+
+  if (route.name === 'ToReadScreen') {
+    return <ToReadList data={storyData || []} />;
+  } else if (route.name === 'PodcastScreen') {
+    return <PodcastList data={podcastData || []} />;
+  } else {
+    return <MusicList data={musicData || []} />;
+  }
+};
```
Comment (BrandonVG) [inline languageApp/src/components/CardList/CardList.tsx:57]: The if-else if-else block is a nice solution, but be aware that if options get increased, readability could suffer, and performance might be impacted due to checking each condition, if you expect more options and the logic isn't complex, consider using a switch or refactoring with an object {route.name: component} with return object[route.name] to improve readability

**Review by BrandonVG — APPROVED** [review]

**Review by AMCernas — APPROVED** [review]
Good job!

**Review by montsegv-2 — APPROVED** [review]
Nice!

**Pull request merged**